### PR TITLE
Re-uploading dataset.rst and updating function.rst 

### PR DESCRIPTION
**Pull request by Jhayford**

### CODE REVIEW

File: docs/demos/function.rst [hunk 1]
```diff
@@ -10,8 +10,10 @@ Function approximation
    :maxdepth: 1
 
    function/func
+   funtion/dataset
```
Comment (lululxvi) [inline docs/demos/function.rst:13]: "funtion" --- typo. Be careful.

File: docs/demos/function.rst [hunk 2]
```diff
@@ -10,8 +10,10 @@ Function approximation
...
+
```
Comment (lululxvi) [inline docs/demos/function.rst:14]: Delete empty lines 14 and 15.

File: docs/demos/function/dataset.rst [hunk 3]
```diff
@@ -0,0 +1,74 @@
+Approximating a Function From a Dataset
```
Comment (lululxvi) [inline docs/demos/function/dataset.rst:1]: Learning a function from a dataset

File: docs/demos/function/dataset.rst [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    fname_test = "../dataset/dataset.test"
+
+The variables fname_train and fname_test are used to to import the dataset and recall the dataset later in the code. 
+
+The next step is to define both fname_train and fname_test and standardize it in an appropriate form
+
+.. code-block:: python
+
+    data = dde.data.DataSet(
+    fname_train=fname_train,
```
Comment (lululxvi) [inline docs/demos/function/dataset.rst:37]: Wrong indent. Please follow the format in the code.

File: docs/demos/function.rst (hunk 1 above)
Comment (Jhayford) [reply docs/demos/function.rst]: Yes. Thank you. Silly mistake

File: docs/demos/function/dataset.rst (hunk 4 above)
Comment (Jhayford) [reply docs/demos/function/dataset.rst]: I totally missed that the indentations were not copied with the code. I fixed it. Thank you

File: docs/demos/function/dataset.rst [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+.. code-block:: python
+
+    data = dde.data.DataSet(
+        fname_train=fname_train,
+        fname_test=fname_test,
+        col_x=(0,),
+        col_y=(1,),
+        standardize=True,
+)
```
Comment (lululxvi) [inline docs/demos/function/dataset.rst:42]: ")": wrong indent.

File: docs/demos/function/dataset.rst [hunk 6]
```diff
@@ -0,0 +1,75 @@
+Learning a Function From a Dataset
+=======================================
+
+Problem setup
+-------------
+
+This document explains how to train a model based on a dataset:
```
Comment (lululxvi) [inline docs/demos/function/dataset.rst:7]: What you wrote is not problem setup.

**Comment by lululxvi** [comment]
You can look at the generated webpage at: https://deepxde--1038.org.readthedocs.build/en/1038/demos/function/dataset.html

File: docs/demos/function/dataset.rst [hunk 7]
```diff
@@ -0,0 +1,75 @@
+Learning a Function From a Dataset
```
Comment (lululxvi) [inline docs/demos/function/dataset.rst:1]: Use exactly "Learning a function from a dataset"

File: docs/demos/function/dataset.rst [hunk 8]
```diff
@@ -0,0 +1,73 @@
+Learning a function from a dataset
+=======================================
+
+Problem setup
+-------------
+
+We will learn how to approximate a function from a dataset
```
Comment (lululxvi) [inline docs/demos/function/dataset.rst:7]: Be specific. Assume you would like to give this problem as a HW to the students. What would you write for the problem?

File: docs/demos/function/dataset.rst [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+        fname_train=fname_train,
+        fname_test=fname_test,
+        col_x=(0,),
+        col_y=(1,),
+        standardize=True,
+    )
+
+After defining the dataset, the specifics of the model are defined. 
+The first line defines the layout of the inputs(dataset) the hidden layers used to train the model and the output.
+The next line specifies the activation function used ``tanh`` and the initializer as ``Glorot uniform``
```
Comment (lululxvi) [inline docs/demos/function/dataset.rst:44]: Miss "." at the end.

File: docs/demos/function/dataset.rst [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+This is a fully connected neural network of depth 4 and a width of 50
```
Comment (lululxvi) [inline docs/demos/function/dataset.rst:45]: "This" is which?

File: docs/demos/function/dataset.rst (hunk 8 above)
Comment (Jhayford) [reply docs/demos/function/dataset.rst:7]: I will make some changes with something specific but I am not sure what the expectation is. My document has no specific functions or mathematical problem set up. Please let me know if it is still insufficient.

File: docs/demos/function/dataset.rst [hunk 11]
```diff
@@ -0,0 +1,72 @@
+Learning a function from a dataset
+=======================================
+
+Problem setup
+-------------
+
+We will learn a function from a dataset. The dataset used to train the model is dataset.train and the dataset used to test the model is dataset.test.
```
Comment (lululxvi) [inline docs/demos/function/dataset.rst:7]: Add links to the dataset file, see example https://deepxde.readthedocs.io/en/latest/demos/pinn_forward/burgers.html

File: docs/demos/function/dataset.rst [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+    import deepxde as dde
+
+The next step is to import the dataset needed for the model training.
+
+.. code-block:: python
+
+    fname_train = "../dataset/dataset.train"
+    fname_test = "../dataset/dataset.test"
+
+The variables fname_train and fname_test are used to import the dataset and recall the dataset later in the code.
```
Comment (lululxvi) [inline docs/demos/function/dataset.rst:29]: add `` for variables

File: docs/demos/function/dataset.rst [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+The next line specifies the activation function used ``tanh`` and the initializer as ``Glorot uniform``.
+
+.. code-block:: python
+
+    layer_size = [1] + [50] * 3 + [1]
+    activation = "tanh"
+    initializer = "Glorot normal"
+    net = dde.nn.FNN(layer_size, activation, initializer)
+
+The model can now be built using ``adam`` as an optimizer with a learning rate of 0.001s.
```
Comment (lululxvi) [inline docs/demos/function/dataset.rst:54]: 0.001s --> 0.001

File: docs/demos/function/dataset.rst [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+
+We will learn a function from a dataset. The dataset used to train the model can be found `here <https://github.com/lululxvi/deepxde/blob/master/examples/dataset/dataset.train>`_. and the dataset used to test the model can be found `here <https://github.com/lululxvi/deepxde/blob/master/examples/dataset/dataset.test>`_.
+
+
+Implementation
+--------------
+
+A step by step description of how to implement this code is written below.
+
+First, install the backend support needed for DeepXDE: Tensorflow and Pytorch, before installing DeepXDE.
```
Comment (lululxvi) [inline docs/demos/function/dataset.rst:15]: Delete "Tensorflow and Pytorch"

File: docs/demos/function/dataset.rst [hunk 15]
```diff
@@ -0,0 +1,73 @@
...
+Import the DeepXDE Library used for this project as described below.
```
Comment (lululxvi) [inline docs/demos/function/dataset.rst:16]: "Library" --> "library"

File: docs/demos/function/dataset.rst [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+    data = dde.data.DataSet(
+        fname_train=fname_train,
+        fname_test=fname_test,
+        col_x=(0,),
+        col_y=(1,),
+        standardize=True,
+    )
+
+After defining the dataset, the specifics of the model are defined. 
+The first line defines the layout of the inputs(dataset) the hidden layers used to train the model and the output.
```
Comment (lululxvi) [inline docs/demos/function/dataset.rst:44]: "the layout of the inputs(dataset) the hidden layers used to train the model and the output" --> "the layout of the network size used to train the model"

**Pull request merged**